### PR TITLE
[Vue] migrate tracking code generation controllers to Vue

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of https://github.com/matomo-org/matomo/pull/18549.**

Changes:
* Migrate the js tracking code generator controller and twig content.
* Migrate the image tracking code generator controller and twig content.
* Add some extra exports to CoreHome.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
While clicking through the UI I found one minor issue. Not sure if that was the case before, but I guess it should be easy to fix:

When you choose a goal for image tracking and afterwards switch the site in the selector to a site where a goal with that id doesn't exist, the goal selector switches to `None`, but the tracking code still includes `idGoal=x` for the previously selected goal.

File: plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+  <ContentBlock
+    anchor="javaScriptTracking"
+    :content-title="translate('CoreAdminHome_JavaScriptTracking')"
+  >
+    <div id="js-code-options">
+      <p>
+        {{ translate('CoreAdminHome_JSTrackingIntro1') }}
+        <br/><br/>
+        {{ translate('CoreAdminHome_JSTrackingIntro2') }}
+        <span v-html="jsTrackingIntro3a"></span> {{ translate('CoreAdminHome_JSTrackingIntro3b') }}
```
Comment (sgiehl) [inline plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue]: I guess the translated text of `CoreAdminHome_JSTrackingIntro3b` is not auto escaped here, causing the escaped `</head>` to be double escaped. This is also visible in the UI screenshot failure.

**Comment by diosmosis** [comment]
> When you choose a goal for image tracking and afterwards switch the site in the selector to a site where a goal with that id doesn't exist, the goal selector switches to None, but the tracking code still includes idGoal=x for the previously selected goal.

Should be fixed.

File: plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+  <ContentBlock
+    anchor="javaScriptTracking"
+    :content-title="translate('CoreAdminHome_JavaScriptTracking')"
+  >
+    <div id="js-code-options">
+      <p>
+        {{ translate('CoreAdminHome_JSTrackingIntro1') }}
+        <br/><br/>
+        {{ translate('CoreAdminHome_JSTrackingIntro2') }}
+        <span v-html="jsTrackingIntro3a"></span>
```
Comment (sgiehl) [inline plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue:17]: There seems to be a space missing between both texts now.
```suggestion
        <span v-html="jsTrackingIntro3a"></span>&nbsp;
```

File: plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+        {{ translate('CoreAdminHome_JSTracking_MergeAliasesDesc', currentSiteAlias) }}
+      </div>
+
+      <Field
+        uicontrol="checkbox"
+        name="javascript-tracking-all-aliases"
+        :model-value="trackAllAliases"
+        @update:model-value="trackAllAliases = $event; updateTrackingCode()"
+        :disabled="isLoading"
+        :title="`${translate('CoreAdminHome_JSTracking_MergeAliases')} ${site.name}`"
```
Comment (sgiehl) [inline plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue]: `site.name` needs to be encoded in some way. As visible on the UI screenshots a html would currently not be escaped correctly. Same applies for the other usage of `site.name`

File: plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+        'y.<span class=\'current-site-host\'></span>',
+      );
+    },
+    learnMoreText() {
+      const subdomainsLink = 'https://developer.matomo.org/guides/tracking-javascript-guide'
+        + '#measuring-domains-andor-sub-domains';
+      return translate(
+        'General_LearnMore',
+        ` (<a href="${subdomainsLink}" rel="noreferrer noopener" target="_blank">`,
+        '</a>',
```
Comment (sgiehl) [inline plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue]: The closing bracket got lost.
```suggestion
        '</a>)',
```

File: plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+        '<a rel="noreferrer noopener" target="_blank" '
+        + 'href="https://developer.matomo.org/guides/tracking-javascript-guide">',
+        '</a>',
+      );
+    },
+    mergeSubdomainsDesc() {
+      return translate(
+        'CoreAdminHome_JSTracking_MergeSubdomainsDesc',
+        'x.<span class=\'current-site-host\'></span>',
+        'y.<span class=\'current-site-host\'></span>',
```
Comment (sgiehl) [inline plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue]: It seems those span were used before to insert the current host, upon site change. Seems that doesn't work anymore.

**Comment by diosmosis** [comment]
@sgiehl should be fixed

**Review by sgiehl — APPROVED** [review]
works now 👍

**Pull request merged**